lib/bosh/BoshDirectorClient.js

```diff
@@ -156,7 +156,21 @@ class BoshDirectorClient extends HttpClient {
 
   makeRequest(requestDetails, expectedStatusCode, deploymentName) {
     return this.getDirectorConfig(deploymentName)
-      .then(directorConfig => this.makeRequestWithConfig(requestDetails, expectedStatusCode, directorConfig));
+      .then(directorConfig => this.makeRequestWithConfig(requestDetails, expectedStatusCode, directorConfig))
+      .catch(NotFound, (err) => {
+        logger.debug('Going to delete the entry from cache for the deployment:', deploymentName);
+        this.deleteCacheEntry(deploymentName);
+        return Promise
+          .map(config.directors, (directorConfig) => this.checkAndUpdateCache(deploymentName, directorConfig)
```

Comment:
Dont invoke this new method, rather invoke `makeRequest` itself (call self) and pass an additional param named `attempt` with value 1. So code should be like the below 
```
makeRequest(requestDetails, expectedStatusCode, deploymentName, attempt){
...
...
(NotFound, (err) => {
      logger.debug('Going to delete the entry from cache for the deployment:', deploymentName);
      this.deleteCacheEntry(deploymentName);
      if(attempt !== undefined){
            throw err;
       }
       return this.makeRequest(requestDetails, expectedStatusCode, deploymentName, 1);
    });
``` 

The idea here is if deployment is not found, you invoke it again by clearing cache. Also  `getDirectorConfig` method must be enhanced to update cache at deployment level rather than removing all contents of cache and reloading entire cache contents. [Comments provided for the below method should give details on how to fix `getDirectorConfig`]